bindings/python/setup.py

```diff
@@ -2,15 +2,69 @@
 # encoding: utf-8
 
 from setuptools import setup, Extension
+from pathlib import Path
+import os
+import shutil
+import sys
+import subprocess
+from distutils.command.build import build
+from distutils.command.sdist import sdist
+from setuptools.command.bdist_egg import bdist_egg
 
+ROOT_DIR = Path(os.path.realpath(__file__)).parent
+BUILD_DIR = Path(ROOT_DIR) / "python_build" / "usr"
 
-rust_module = Extension('udbserver',
-                           sources=['udbserver.c'],
-                           libraries=['udbserver'],
-                           )
+VERSION = "0.1"
+
+def cargo_clean():
+    args = ["cargo", "clean"]
+    subprocess.check_output(args, cwd=ROOT_DIR)
+
+def cargo_cinstall():
+    args = ["cargo", "cinstall"]
+    if not os.getenv("DEBUG", ""):
+        args += ["--release"]
+    args += ["--prefix", "/usr", "--destdir", "python_build"]
+    subprocess.check_output(args, cwd=ROOT_DIR)
+
+
+class custom_sdist(sdist):
+    def run(self):
+        return sdist.run(self)
+
+class custom_build(build):
+    def run(self):
+        cargo_clean()
+        cargo_cinstall()
+        return build.run(self)
+
+class custom_bdist_egg(bdist_egg):
+    def run(self):
+        self.run_command('build')
+        return bdist_egg.run(self)
+
+cmdclass = {}
+cmdclass['build'] = custom_build
+cmdclass['sdist'] = custom_sdist
+cmdclass['bdist_egg'] = custom_bdist_egg
+
+extra_link_args = {
+    "linux" : ["-Wl,--no-as-needed", "-l:libunicorn.so"],
+    "linux2" : ["-Wl,--no-as-needed", "-l:libunicorn.so"],
+    "darwin": ["-lunicorn"]
+}
+
+
+rust_module = Extension('udbserver_rust',
```

Comment:
```suggestion
rust_module = Extension('udbserver',
```
This could be `udbserver`? It seems that `PyInit_udbserver_rust` is enough.

Reply:
no, note there is already a wrapper existing to import unicorn and provide a function definition for easy use here and thus it's renamed.